invokeai/app/services/model_install/model_install_default.py

```diff
@@ -0,0 +1,401 @@
+"""Model installation class."""
+
+import threading
+from hashlib import sha256
+from pathlib import Path
+from queue import Queue
+from random import randbytes
+from shutil import copyfile, copytree, move, rmtree
+from typing import Any, Dict, List, Optional, Set, Union
+
+from invokeai.app.services.config import InvokeAIAppConfig
+from invokeai.app.services.events import EventServiceBase
+from invokeai.app.services.model_records import DuplicateModelException, ModelRecordServiceBase, UnknownModelException
+from invokeai.backend.model_manager.config import (
+    AnyModelConfig,
+    BaseModelType,
+    InvalidModelConfigException,
+    ModelType,
+)
+from invokeai.backend.model_manager.hash import FastModelHash
+from invokeai.backend.model_manager.probe import ModelProbe
+from invokeai.backend.model_manager.search import ModelSearch
+from invokeai.backend.util import Chdir, InvokeAILogger
+
+from .model_install_base import (
+    InstallStatus,
+    ModelInstallJob,
+    ModelInstallServiceBase,
+    ModelSource,
+    UnknownInstallJobException,
+)
+
+# marker that the queue is done and that thread should exit
+STOP_JOB = ModelInstallJob(source="stop", local_path=Path("/dev/null"))
+
+
+class ModelInstallService(ModelInstallServiceBase):
+    """class for InvokeAI model installation."""
+
+    _app_config: InvokeAIAppConfig
+    _record_store: ModelRecordServiceBase
+    _event_bus: Optional[EventServiceBase] = None
+    _install_queue: Queue[ModelInstallJob]
+    _install_jobs: Dict[ModelSource, ModelInstallJob]
```

Comment:
A little earlier today I checked in a version in which `ModelSource` is a discriminated Union of types that derive from a base class that has a `__hash__()` method. So that parts works.

But the second point is problematic. Using a regular list won't be a problem, but I think I have to change `get_job(source)->ModelInstallJob` to `get_job(source)->List[ModelInstallJob]`, since there won't be a one-to-one relationship between a source and an install job.

Reply:
I've implemented the requested change. `_install_jobs` is now a `List[ModelInstallJob]`